### PR TITLE
Fix incorrect model output due to MPS nn.Linear bug.

### PR DESCRIPTION
**Pull request by oscarkey**
On at least MacOS 26 + M1, nn.Linear can have the wrong output when bias is enabled: https://github.com/pytorch/pytorch/issues/188438
This PR tests if we hit the bug. If so, it replaces nn.Linear layers in the model with a decomposed version.

Switch the runner back to macos-latest (which is either macos-15 or macos-26 on M1).

Testing: in [this run](https://github.com/PriorLabs/TabPFN/actions/runs/28606803966/job/84829060784) I forced it to macos-26, the tests pass

Fixes PRI-331

### CODE REVIEW

File: src/tabpfn/architectures/shared/workaround_mps_linear_bug.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    with torch.no_grad():
+        torch.manual_seed(0)
+        cpu = nn.Linear(2, 192, bias=True)
+        mps = nn.Linear(2, 192, bias=True).to("mps")
+        mps.load_state_dict(cpu.state_dict())
+        x = torch.randn(80, 1, 2)
+        rel = (
+            (cpu(x) - mps(x.to("mps")).cpu()).abs().max() / cpu(x).abs().max()
+        ).item()
+    return rel > 1e-3
```
[the remark this answers is not in the document]
Comment (oscarkey) [reply src/tabpfn/architectures/shared/workaround_mps_linear_bug.py:45]: I fixed (1).
2) I'll add the exception handling
3) Seems so unlikely we can ignore?

[the remark this answers is not in the document]
Comment (arthur-priorlabs) [reply src/tabpfn/architectures/shared/workaround_mps_linear_bug.py:45]: For 3. I'd just divide by `cpu(x).abs().max() + 1e-6`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,9 +53,9 @@ jobs:
           - os: windows-latest
             python-version: "3.10"
             dependency-set: lowest-direct
-          # The MacOS tests are failing on macos-26, so pin to 15 for now. See
-          # https://linear.app/priorlabs/issue/PRI-331/ci-failing-when-runner-is-macos-26
-          - os: macos-15
+          # TEMPORARY: forced to macos-26 to test the MPS Linear workaround on the
+          # affected runner. Revert to macos-latest before merging.
+          - os: macos-26
```
Comment (arthur-priorlabs) [inline .github/workflows/ci.yml]: revert before merging? 😀

File: src/tabpfn/architectures/shared/workaround_mps_linear_bug.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    nn.Linear modules are affected; bias=False layers are left untouched (they are
+    not affected by the bug). Mutates and returns model.
+    """
+    if not _mps_linear_bias_is_buggy():
+        return model
+    for module in model.modules():
+        if type(module) is nn.Linear and module.bias is not None:
+            # Same memory layout and parameters, so re-typing in place is safe and
+            # preserves weights, buffers, hooks, and state_dict compatibility.
+            module.__class__ = MpsSafeLinear
```
Comment (arthur-priorlabs) [inline src/tabpfn/architectures/shared/workaround_mps_linear_bug.py:78]: neat trick!

**Review by arthur-priorlabs — APPROVED** [review]
LGTM, thanks for the fix!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (oscarkey) [reply .github/workflows/ci.yml]: oops, done!

**Comment by oscarkey** [comment]
thanks @arthur-priorlabs , would you take one more quick look? I moved the conversion to inside the `.to()` function so it only runs if MPS is selected. This solves Claude's concern that the detection code uses mps even if mps is not selected.

**Comment by arthur-priorlabs** [comment]
This looks good, just one concern: it doesn't seem like your fix will fire unless the model is moved to MPS via our inference engine, i.e. plain PyTorch `model.to("mps")` will still produce errors.

Maybe not super relevant though, so up to you.

**Comment by oscarkey** [comment]
I think that's fine, we only support people changing the device by calling `TabPFNClassifier/Regressor.to()`, which always goes via this code path.

**Review by arthur-priorlabs — APPROVED** [review]

File: src/tabpfn/architectures/shared/workaround_mps_linear_bug.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (oscarkey) [reply src/tabpfn/architectures/shared/workaround_mps_linear_bug.py:45]: I think it's so unlikely (esp with the fixed seed) that I prefer the shorter code.
